BS3 -> BS4 Migration Review in Dev

ctrepka opened the pull request:
Migrating from Bootstrap 3 to Bootstrap 4

ctrepka committed:
F&R BS3 classes, install BS4, carousel refactor, nav refactor

Refactored:                                                                       
.col-*-offset-*
.col-*-push-*
.col-*-pull-*
.panel
.panel-heading
.panel-title
.panel-body
.panel-footer
.panel-primary
.panel-success
.panel-info
.panel-warning
.panel-danger
.well
.img-responsive

as well as portions of navbar.njk and mainfeature.njk

jwhaney committed:
Merge branch 'develop'

ctrepka committed:
More bs4 class substitutions, removed url regex from site.js

edited the following classes to bs4 equivalents:                                                                                        .img-responsive
.img-circle
.img-rounded
.form-horizontal
.radio
.checkbox
.input-lg
.input-sm
.control-label
.table-condensed
.pagination > li
.pagination > li > a
.item
.help-block
.pull-right
.pull-left
.center-block
.hidden-xs
.hidden-sm
.hidden-md
.hidden-lg
.visible-xs
.visible-sm
.visible-md
.visible-lg
.label
.badge
.col-xs-*

removed site.js regex line that transformed relative path hrefs to absolute path hrefs to maintain functionality of bs4 carousel next and prev buttons

ctrepka committed:
Refactored navigation, sub-navigation, header margin, and row/col markup

- Refactored markup broke column stacking (bs4 requires columns to be contained by .row class to stack as expected)
- Refactored navigation and subnavigations across site to highlight active page in nav
- Refactored nav and subnav to stack properly at variety of breakpoints
- Added nav classes nav-item and nav-link to all navigations

ctrepka committed:
Navigation fix for bs4, collapse fix, padding fix,  glyphicon F&R

-More fixes for columns collapse
-Made header sizes smaller to reflect previous bs3 header sizes
-Fixed navigation collapse issues for main navigation menu
-Fixed secondary navs collapse padding (.5rem on small screens < 574px)
-Replaced glyphicons with font awesome
-Replaced some icons that do not transfer directly from .glyphicons .glyphicon-*-* -> .fa .fa-*-*

ctrepka committed:
QA and markup changes to past events

Cleanup and QA, mostly small changes. More columns moved into rows, some responsive fixes, etc.

ctrepka committed:
Merge branch 'master' into feature/bs4-migration

adambreznicky commented:
manually reviewed the changes side-by-side with the current prod. absolutely awesome!!! in many cases, the mobile sytling was fixed with padding (etc.) as needed. excellent job :)

the only issue i found at all:
on the /maps page, it successfully retrieves the map collections from the api and displays the options but the "Expand Available Maps" button on each is supposed to accordion and display the download links on-click. this onlick and list display doesn't seem to work. the text updates inside the button itself but the list doesn't display.

@ctrepka if you could fix the minor /maps bug and push it on this branch, it'll automatically roll into this PR. at that time, i'll go a super quick glance to verify and then we can close this PR out. get these updates into production.
fantastic job knocking this issue out. it is beyond appreciated and a solid job well done.

adambreznicky commented:
update::: it seems like the /maps "Expand Available Maps" button is working inconsistenly.
it works for the collections:
"TWDB - Groundwater"
"TWDB - Groundwater Management Areas (GMA)"
"TWDB - Regional Water Project and Development (RWPD)"
"TWDB - Supplementals"
"TWDB - Water Well Grid Map Series"

the other 5 don't though. hmmmmmm, very interesting
must be something particular in the .js file that is failing when dismantling the API response to build these HTML elements.

ctrepka commented:
> ust be something particular in the .js file that is failing when dismantling the API response to build these HTML elements.

Hey @adambreznicky just seeing these comments somehow.. Thanks for catching this one! I'll jump on this now.

It seems like the button may be assigned the wrong ID or something. When I examine the rendered HTML in the browser, the hidden block content is rendered, but the "collapse" class that toggles the display of that hidden block doesn't change onClick for those five collections you mentioned. If I erase the "collapse" class from the inspector in the browser, the contents display. Guessing it was some weird manipulation with the regEx I did by accident, or maybe related to our absolute/relative hyperlink issue discussed previously. I'll keep you posted and hopefully it will be a quick fix!

ctrepka committed:
Fixed maps.js (/maps) tables issue

maps.js injected map collection ids + 'Sheets' as html ids. Some started with numbers' causing issues with Bootstrap4 expand animations. Appended 'Sheets' to start of collection ids so that they all begin with letters.
